modelplace_api/visualization.py

```diff
@@ -399,6 +415,73 @@ def draw_segmentation_one_frame(image: np.ndarray, detection: Mask) -> np.ndarra
     return image
 
 
+def draw_instance_segmentation_one_frame(image: np.ndarray, instance_mask: InstanceMask) -> np.ndarray:
+    predicted_classes = [
+        mask.mask["classes"] for mask in instance_mask.masks
+    ]
+    unique_predicted_classes = list(set(elem for sublist in predicted_classes for elem in sublist))
+    classes = [
+        class_name
+        for class_number, class_name in enumerate(instance_mask.classes)
+        if class_number in unique_predicted_classes
+    ]
+    result_mask = None
+    color_boxes = []
+    for instance_number, (mask, box) in enumerate(zip(instance_mask.masks, instance_mask.detections)):
+        # skip background here
+        rle_mask = mask.mask["binary"][-1]
+        color = RGBA_COLORS[instance_number]
+        decoded_mask = decode_coco_rle(rle_mask)
+        idx = decoded_mask == 1
+        result_mask = form_mask(image, idx, color, result_mask)
+        color_boxes.append((box, color))
+    image = add_instance_mask(image, result_mask)
+    for (box, color) in color_boxes:
+        image = add_bbox(image,[box.x1, box.y1, box.x2, box.y2], color)
+    image = add_legend_all_classes(image, classes, instance_segmentation=True)
+    return image
+
+
+def draw_instance_segmentation(image: np.ndarray, instance_mask: InstanceMask) -> np.ndarray:
+    source_image = image.copy()
```

Comment:
Can we call the draw_instance_segmentation_one_frame function here instead of duplicating the code?
Check out other visualizations.

Reply:
It seems like we can't